### DISCUSSION TITLE
changing global variables within on_finish function?

### DISCUSSION THREAD

**Discussion by addisonjadwin** [jspsych/jsPsych#2075] [Q&A] [answered]
Hello,

I've declared the string "outcome" globally here, outside of any functions.

```
var outcome;
```

I then edit the string using responses from a survey multi choice variable, within the on_finish(data) function:(at the emphasized line with **)

```
function showTrialQ(face){
    var q = {
        type: 'survey-multi-choice',
        preamble: "<img src=\'img/faces/"+face['image']+"\'> <h2 style=\"text-align: center;\">",
        button_label: button_str,
        questions: [{
            prompt: "<h2 style=\"text-align: center;\">How much will you give to Player 2?</h2>",
            options: money_option,
            horizontal: true,
            required: true}],
        data: {
            ...
        },
        on_finish: function(data) {
          var amountShared = data.responses[13];
          **outcome = "Player 2 received $" + amountShared;**
          }
        }
    };
    return q;
}
```

But, when I use the "outcome" string here as the prompt to this keyboard response trial, it registers as "undefined" and shows up
on the experiment as such, as if I had never changed it. (this keyboard response trial is pushed to timeline after the multi choice survey).
```
function outcomeFunction(face){
    var reputation = {
        type: "html-keyboard-response",
        stimulus: "<img src=\'img/faces/"+face['image']+"\'> <h2 style=\"text-align: center;\"></h2>",
        **prompt: "<h2 style=\"text-align: center;\">" + outcome,**
        choices: [],
        trial_duration: reputation_duration * 1000
    };
    return reputation;
};
```
Does anyone know what I'm doing wrong? Any help is appreciated, thanks!

**Comment by becky-gilbert** [comment, marked as answer]
Hi @addisonjadwin, the prompt parameter that uses the outcome variable is defined when the page first loads, at which point `outcome` is undefined. So I think all you need to do is turn that prompt parameter into a dynamic parameter (function). That way the parameter value is defined during the experiment. Does that work?

**Reply by addisonjadwin** [reply]
Thanks so much for your help @becky-gilbert , this worked! :)

**Reply by becky-gilbert** [reply]
Great, thanks for the update!
